src/index.js

```diff
@@ -156,9 +156,17 @@ class Griddle extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const { data, pageProperties, sortProperties } = nextProps;
-
-    this.store.dispatch(actions.updateState({ data, pageProperties, sortProperties }));
+    const newState = Object.keys(nextProps).reduce((updatedState, key) => {
+      return this.props[key] !== nextProps[key] ? {
+      ...updatedState,
+      [key]: nextProps[key],
```

Comment:
Mutate accumulator here for performance?

Reply:
Good call! Updated to `_.pickBy`